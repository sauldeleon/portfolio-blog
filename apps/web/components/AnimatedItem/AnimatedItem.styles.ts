import styled, { RuleSet, css } from 'styled-components'

import {
  randomDecimalFromInterval,
  randomIntFromInterval,
} from '@web/utils/random'

export const VerticalMovement = styled.div.attrs<{
  $seed: string
  $customAnimation?: RuleSet<object>
}>(({ $seed }) => ({
  style: {
    '--translateY-begin':
      randomIntFromInterval(-37, 20, `${$seed}-y-begin`) + '%',
    '--translateY-end': randomIntFromInterval(-20, 37, `${$seed}-y-end`) + '%',
    animationDuration:
      randomDecimalFromInterval(2, 10, `${$seed}-item-vertical-duration`) + 's',
  },
}))`
  position: absolute;
  width: 100%;
  height: 100%;
  display: grid;
  place-items: center;
  pointer-events: none;
  animation-timing-function: ease-in-out;
  animation-iteration-count: infinite;
  animation-direction: alternate;
  ${({ $customAnimation, theme }) =>
    $customAnimation ||
    css`
      animation-name: ${theme.animation.verticalMovement};
    `};
`

export const RotationMovement = styled.div.attrs<{
  $customAnimation?: RuleSet<object>
  $seed: string
  $rotate?: boolean
}>(({ $rotate, $seed }) => ({
  style: $rotate
    ? {
        '--rotation':
          'calc(' +
          randomIntFromInterval(1, 2, `${$seed}-rotation-amount`) +
          ' * 360deg)',
        zIndex: randomIntFromInterval(2, 5, `${$seed}-item-z-index`),
        animationDuration:
          randomDecimalFromInterval(4, 6, `${$seed}-item-rotation-duration`) +
          's',
      }
    : undefined,
}))`
  height: var(--itemSize);
  aspect-ratio: 1/1;
  position: absolute;
  pointer-events: none;

  ${({ $rotate, $customAnimation, theme }) =>
    $rotate
      ? $customAnimation ||
        css`
          animation-name: ${theme.animation.rotateMovement};
          animation-timing-function: linear;
          animation-iteration-count: infinite;
        `
      : null}
`
export const StyledExternalLink = styled.a`
  pointer-events: all;
  cursor: pointer;
`

export const ColorSwapping = styled.div.attrs<{
  $seed: string
  $colorSwap?: boolean
}>(({ $seed }) => ({
  style: {
    animationDuration:
      randomDecimalFromInterval(6, 20, `${$seed}-item-horizontal-duration`) +
      's',
    animationDelay:
      randomDecimalFromInterval(1, 20, `${$seed}-item-horizontal-delay`) + 's',
  },
}))`
  ${({ theme, $colorSwap }) => css`
    color: ${theme.colors.yellow};
    ${$colorSwap &&
    css`
      animation-name: ${theme.animation.colorSwap};
      animation-timing-function: linear;
      animation-iteration-count: infinite;
    `}

    svg {
      width: 100%;
      height: 100%;
    }
  `}
`

export const HorizontalMovement = styled.li.attrs<{
  $size: number
  $seed: string
  $customAnimation?: RuleSet<object>
}>(({ $seed }) => ({
  style: {
    animationDuration:
      randomDecimalFromInterval(6, 20, `${$seed}-item-horizontal-duration`) +
      's',
    animationDelay:
      randomDecimalFromInterval(1, 20, `${$seed}-item-horizontal-delay`) + 's',
  },
}))`
  --itemSize: ${({ $size }) => `${$size}px`};
  list-style: none;
  position: absolute;
  transform: translateX(-60%);
  width: 100%;
  height: 100%;
  pointer-events: none;

  ${({ $customAnimation, theme }) =>
    $customAnimation ||
    css`
      animation-name: ${theme.animation.horizontalMovement};
      animation-timing-function: linear;
      animation-iteration-count: infinite;
    `}

  &:hover {
    z-index: 200;
    animation-play-state: paused;
    ${VerticalMovement},${RotationMovement},${ColorSwapping} {
      animation-play-state: paused;
    }
  }

  ${({ theme }) => theme.media.up.md} {
    --itemSize: ${({ $size }) => `${$size + 15}px`};
  }
`
