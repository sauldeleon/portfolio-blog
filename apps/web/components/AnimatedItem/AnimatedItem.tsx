import React, { useId } from 'react'
import { RuleSet } from 'styled-components'

import {
  HorizontalMovement,
  RotationMovement,
  VerticalMovement,
} from './AnimatedItem.styles'

export type CustomAnimation = {
  horizontal?: RuleSet<object>
  vertical?: RuleSet<object>
  rotate?: RuleSet<object>
}

export type AnimationSize = 14 | 20 | 26

export interface AnimatedItemProps {
  path?: string
  rotate?: boolean
  size?: AnimationSize
  customAnimation?: CustomAnimation
}

export function AnimatedItem({
  path,
  rotate,
  customAnimation,
  size = 20,
}: AnimatedItemProps) {
  const theKey = useId()
  return (
    <HorizontalMovement
      role="presentation"
      $size={size}
      $seed={theKey}
      $customAnimation={customAnimation?.horizontal}
    >
      <VerticalMovement
        $seed={theKey}
        $customAnimation={customAnimation?.vertical}
      >
        <RotationMovement
          role="none"
          $rotate={rotate}
          $path={path}
          $seed={theKey}
          $customAnimation={customAnimation?.rotate}
        />
      </VerticalMovement>
    </HorizontalMovement>
  )
}
