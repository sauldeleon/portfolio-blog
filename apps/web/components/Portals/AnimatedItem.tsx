import React from 'react'
import { RuleSet } from 'styled-components'

import {
  HorizontalMovement,
  RotationMovement,
  VerticalMovement,
} from './AnimatedItem.styles'
import { randomIntFromInterval } from './helpers'

export type AnimatedItemSeeds = {
  keyframe: string
  verticalStartPoint: string
  horizontalDuration: string
  horizontalDelay: string
  verticalZIndex: string
  verticalDuration: string
  verticalDelay: string
  verticalRange: string
  verticalColor: string
  rotationZIndex: string
  rotationColor: string
  rotationDuration: string
  rotationDelay: string
}

export type CustomAnimation = {
  horizontal?: RuleSet<object>
  vertical?: RuleSet<object>
  rotate?: RuleSet<object>
}

export interface AnimatedItemProps {
  id: number
  seeds: AnimatedItemSeeds
  parentWidth: number
  parentHeight: number
  path?: string
  rotate?: boolean
  size?: number
  customAnimation?: CustomAnimation
}

export function AnimatedItem({
  id,
  seeds,
  parentWidth,
  parentHeight,
  path,
  rotate,
  customAnimation,
  size = 50,
}: AnimatedItemProps) {
  const verticalStartPoint = randomIntFromInterval(
    0,
    parentHeight - 2 * size,
    seeds.verticalStartPoint
  )

  const innerElement: React.ReactNode = rotate ? (
    <VerticalMovement
      $customAnimation={customAnimation?.vertical}
      $id={id}
      $parentHeight={parentHeight}
      $rotate={rotate}
      $size={size}
      $verticalStartPoint={verticalStartPoint}
      $colorSeed={seeds.verticalColor}
      $delaySeed={seeds.verticalDelay}
      $durationSeed={seeds.verticalDuration}
      $verticalRangeSeed={seeds.verticalRange}
      $zIndexSeed={seeds.verticalZIndex}
    >
      <RotationMovement
        $customAnimation={customAnimation?.rotate}
        $id={id}
        $path={path}
        $colorSeed={seeds.rotationZIndex}
        $delaySeed={seeds.rotationDelay}
        $durationSeed={seeds.rotationDuration}
        $zIndexSeed={seeds.rotationZIndex}
      />
    </VerticalMovement>
  ) : (
    <VerticalMovement
      $customAnimation={customAnimation?.vertical}
      $id={id}
      $parentHeight={parentHeight}
      $path={path}
      $size={size}
      $verticalStartPoint={verticalStartPoint}
      $colorSeed={seeds.verticalColor}
      $delaySeed={seeds.verticalDelay}
      $durationSeed={seeds.verticalDuration}
      $verticalRangeSeed={seeds.verticalRange}
      $zIndexSeed={seeds.verticalZIndex}
    />
  )

  return (
    <HorizontalMovement
      $customAnimation={customAnimation?.horizontal}
      $id={id}
      $parentWidth={parentWidth}
      $size={size}
      $verticalStartPoint={verticalStartPoint}
      $delaySeed={seeds.horizontalDelay}
      $durationSeed={seeds.horizontalDuration}
    >
      {innerElement}
    </HorizontalMovement>
  )
}
