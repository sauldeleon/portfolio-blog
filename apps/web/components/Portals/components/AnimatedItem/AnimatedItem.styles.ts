import { RuleSet, css, styled } from 'styled-components'

import {
  randomDecimalFromInterval,
  randomIntFromInterval,
} from '@web/utils/random'

export const HorizontalMovement = styled.div<{
  $size: number
  $seed: string
  $customAnimation?: RuleSet<object>
}>`
  --itemSize: ${({ $size }) => `${$size}%`};
  position: absolute;
  transform: translateX(-60%);
  width: 100%;
  height: 100%;

  ${({ $customAnimation, $seed }) =>
    $customAnimation
      ? $customAnimation
      : css`
          animation: ${randomDecimalFromInterval(
              6,
              20,
              `${$seed}-item-horizontal-duration`
            )}s
            horizontal-movement
            ${randomDecimalFromInterval(
              1,
              5,
              `${$seed}-item-horizontal-delay`
            )}s
            linear infinite;
        `}
`

export const VerticalMovement = styled.div<{
  $seed: string
  $customAnimation?: RuleSet<object>
}>`
  position: absolute;
  width: 100%;
  height: 100%;
  display: grid;
  place-items: center;
  ${({ $customAnimation, $seed }) =>
    $customAnimation
      ? $customAnimation
      : css`
          --translateY-begin: ${randomIntFromInterval(
            -37,
            20,
            `${$seed}-y-begin`
          )}%;
          --translateY-end: ${randomIntFromInterval(
            -20,
            37,
            `${$seed}-y-end`
          )}%;
          animation: ${randomDecimalFromInterval(
              2,
              10,
              `${$seed}-item-vertical-duration`
            )}s
            vertical-movement ease-in-out alternate infinite;
        `};
`

export const RotationMovement = styled.div<{
  $customAnimation?: RuleSet<object>
  $seed: string
  $rotate?: boolean
}>`
  z-index: ${({ $seed }) =>
    randomIntFromInterval(2, 5, `${$seed}-item-z-index`)};
  height: var(--itemSize);
  aspect-ratio: 1/1;
  position: absolute;

  ${({ $rotate, $customAnimation, $seed }) =>
    $rotate
      ? $customAnimation
        ? $customAnimation
        : css`
            --rotation: calc(
              ${randomIntFromInterval(1, 2, `${$seed}-rotation-amount`)} *
                360deg
            );
            animation: ${randomDecimalFromInterval(
                4,
                6,
                `${$seed}-item-rotation-duration`
              )}s
              rotate-movement linear infinite;
          `
      : css``}
`

export const StyledSVGWrapper = styled.div<{
  $seed: string
  $colorSwap?: boolean
}>`
  ${({ theme, $seed, $colorSwap }) => css`
    svg {
      width: 100%;
      height: 100%;

      ${$colorSwap &&
      css`
        color: ${theme.colors.yellow};
        animation: ${randomDecimalFromInterval(
            6,
            20,
            `${$seed}-item-horizontal-duration`
          )}s
          color-swap
          ${randomDecimalFromInterval(1, 5, `${$seed}-item-horizontal-delay`)}s
          linear infinite;
      `}
    }
  `}
`
