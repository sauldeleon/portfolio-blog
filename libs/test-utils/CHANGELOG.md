# Changelog Test utils

## [0.2.3](https://github.com/sauldeleon/portfolio-blog/compare/test-utils-0.2.2...test-utils-0.2.3) (2024-01-08)

## [0.2.2](https://github.com/sauldeleon/portfolio-blog/compare/test-utils-0.2.1...test-utils-0.2.2) (2023-09-26)

## [0.2.1](https://github.com/sauldeleon/portfolio-blog/compare/test-utils-0.2.0...test-utils-0.2.1) (2023-09-14)

# [0.2.0](https://github.com/sauldeleon/portfolio-blog/compare/test-utils-0.1.0...test-utils-0.2.0) (2023-09-05)

### Bug Fixes

- reduced duplicated lines ([f51168a](https://github.com/sauldeleon/portfolio-blog/commit/f51168aefe025a5a65ffdb2dc22f7a16b863406e))
- rename commit message format ([e1e2ab4](https://github.com/sauldeleon/portfolio-blog/commit/e1e2ab404bbd2c32f3508d1ed8197b3fbff93cb9))

### Features

- deleted changelogs ([e9be338](https://github.com/sauldeleon/portfolio-blog/commit/e9be33836ee47b6505ad94d21f4be21855a7fa0d))

# 0.1.0 (2023-09-02)

### Bug Fixes

- rename commit message format ([e1e2ab4](https://github.com/sauldeleon/portfolio-blog/commit/e1e2ab404bbd2c32f3508d1ed8197b3fbff93cb9))

### Features

- add contact initial page ([fea862b](https://github.com/sauldeleon/portfolio-blog/commit/fea862b3507268013149050ae3f05fa517e7b3d9))
- add homepage ([60523aa](https://github.com/sauldeleon/portfolio-blog/commit/60523aa31ec088d318745ef0c757b4cbe656f0d4))
- deleted changelogs ([e9be338](https://github.com/sauldeleon/portfolio-blog/commit/e9be33836ee47b6505ad94d21f4be21855a7fa0d))
- improved a11y and styles ([873f656](https://github.com/sauldeleon/portfolio-blog/commit/873f656587f50df8e29ad526fdc68fea0b2c9784))
- toggle language button ([afd97e9](https://github.com/sauldeleon/portfolio-blog/commit/afd97e95beac0f70ed643c753e71db2ca014054b))
