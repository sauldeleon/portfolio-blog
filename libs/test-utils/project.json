{
  "name": "test-utils",
  "$schema": "../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "libs/test-utils/src",
  "projectType": "library",
  "tags": [],
  "targets": {
    "version": {
      "executor": "@jscutlery/semver:version",
      "options": {
        "changelogHeader": "# Changelog Test utils \n",
        "baseBranch": "main",
        "commitMessageFormat": "chore(${projectName}): release: v${version}",
        "push": false
      }
    },
    "lint": {
      "executor": "@nx/linter:eslint",
      "outputs": ["{options.outputFile}"],
      "options": {
        "lintFilePatterns": ["libs/test-utils/**/*.{ts,tsx,js,jsx}"]
      }
    },
    "test": {
      "executor": "@nx/jest:jest",
      "outputs": ["{workspaceRoot}/coverage/{projectRoot}"],
      "options": {
        "jestConfig": "libs/test-utils/jest.config.ts",
        "passWithNoTests": true
      },
      "configurations": {
        "ci": {
          "ci": true,
          "codeCoverage": true
        }
      }
    }
  }
}
