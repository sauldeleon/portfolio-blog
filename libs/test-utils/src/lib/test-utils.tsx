import { QueryClient, QueryClientProvider } from '@tanstack/react-query'
import { Matcher } from '@testing-library/dom/types/matches'
import {
  RenderOptions,
  RenderResult,
  queryAllByAttribute as _queryAllByAttribute,
  queryByAttribute as _queryByAttribute,
  act,
  render,
  screen,
} from '@testing-library/react'
import { RouterContext } from 'next/dist/shared/lib/router-context'
import Router, { NextRouter } from 'next/router'
import { ReactNode, useEffect, useState } from 'react'
import { ThemeProvider } from 'styled-components'

import { LanguageContextProvider } from '@sdlgr/i18n-config'
import { mainTheme } from '@sdlgr/main-theme'

export type MatcherCustomFnParams = {
  content: string
  element: Element | null
}

/* istanbul ignore next */
export const renderWithTheme = (children: ReactNode): RenderResult => {
  return render(<ThemeProvider theme={mainTheme}>{children}</ThemeProvider>)
}

/* istanbul ignore next */
export function RenderProviders({ children }: { children: ReactNode }) {
  const queryClient = new QueryClient({
    defaultOptions: {
      queries: {
        retry: false,
      },
    },
  })

  return (
    <QueryClientProvider client={queryClient}>
      <LanguageContextProvider value={{ language: 'en' }}>
        <ThemeProvider theme={mainTheme}>{children}</ThemeProvider>
      </LanguageContextProvider>
    </QueryClientProvider>
  )
}

/* istanbul ignore next */
export const renderApp = (
  children: ReactNode,
  options?: Omit<RenderOptions, 'queries'>
) => {
  return render(<RenderProviders>{children}</RenderProviders>, options)
}

/**
 * Generic function to get element by attribute and value
 * @example queryByAttribute('id')(baseElement, 'element-id')
 * @param {string} attr
 * @return {any}
 */
export function queryByAttribute(attr: string) {
  return _queryByAttribute.bind(null, attr)
}

/**
 * Generic function to get all elements by attribute and value
 * @example queryByAttribute('class')(baseElement, 'element-class')
 * @param {string} attr
 * @returns {any}
 */
export function queryAllByAttribute(attr: string) {
  return _queryAllByAttribute.bind(null, attr)
}

/**
 * Generic function to get element by attribute and value
 * @example getByAttribute('id')(baseElement, 'element-id')
 * @param {string} attr
 * @return {any}
 */
export function getByAttribute(attr: string) {
  return _queryByAttribute.bind(null, attr)
}

/**
 * Generic function to get all elements by attribute and value
 * @example getByAttribute('class')(baseElement, 'element-class')
 * @param {string} attr
 * @returns {any}
 */
export function getAllByAttribute(attr: string) {
  return _queryAllByAttribute.bind(null, attr)
}

/**
 * Shortcut version for queryByAttribute, but specific for id
 * @param {HTMLElement} container
 * @param {string} id
 * @return {any}
 */
export function getById(container: HTMLElement, id: string) {
  return queryByAttribute('id')(container, id) as Element
}

/**
 * Shortcut version for queryByAttribute, but specific for class name
 * @param {HTMLElement} container
 * @param {string} className
 * @return {any}
 */
export function getByClassName(container: HTMLElement, className: string) {
  return queryByAttribute('class')(container, className) as Element
}

function matcherCustomFn(
  tag: string,
  matcher: Matcher,
  { element, content }: MatcherCustomFnParams
) {
  const isTag = element?.tagName.toUpperCase() === tag.toUpperCase()
  let matching = false
  if (isTag) {
    matching = content === matcher
    /* istanbul ignore else*/
    if (typeof matcher === 'function') {
      matching = matcher(content, element)
    } else if (matcher instanceof RegExp) {
      matching = matcher.test(content)
    }
  }
  return matching && isTag
}

/**
 * Get 1 element by tag and text
 * @param {string} tag
 * @param {Matcher} matcher
 * @return {any}
 */
export function getByTagAndText(tag: string, matcher: Matcher) {
  return screen.getByText((content, element) =>
    matcherCustomFn(tag, matcher, { content, element })
  )
}

/**
 * Get multiple elements by tag and text
 * @param {string} tag
 * @param {Matcher} matcher
 * @return {any}
 */
export function getAllByTagAndText(tag: string, matcher: Matcher) {
  return screen.getAllByText((content, element) =>
    matcherCustomFn(tag, matcher, { content, element })
  )
}

/**
 * Mock Math.random() to return a specific value
 * @param {number} value
 */
export function mockRandom(value = 0.1234) {
  const mockMath = Object.create(global.Math)
  mockMath.random = () => value
  global.Math = mockMath
}

/**
 * Hook to use in test to ensure "last" render is called
 */
export function useWithRefreshInTest() {
  const [withRefreshInTest, setWithRefreshInTest] = useState(false)

  useEffect(() => {
    // Change state to "rerender" the component
    setWithRefreshInTest(true)
  }, [])

  return withRefreshInTest
}

export { render } from '@testing-library/react'

/* istanbul ignore next */
export const awaitAllAsync = async () =>
  act(() => new Promise((resolve) => setTimeout(resolve, 0)))

/* istanbul ignore next */
export function renderWithNextRouter(
  children: ReactNode,
  options?: Partial<NextRouter>
) {
  return renderWithTheme(
    <RouterContext.Provider value={{ ...Router, ...options }}>
      {children}
    </RouterContext.Provider>
  )
}
