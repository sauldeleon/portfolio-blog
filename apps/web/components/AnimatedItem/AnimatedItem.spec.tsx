/* eslint-disable testing-library/no-node-access */
import { screen } from '@testing-library/react'
import { css } from 'styled-components'

import { renderWithTheme } from '@sdlgr/test-utils'

import { AnimatedItem } from './AnimatedItem'

describe('AnimatedItem', () => {
  it('should render successfully', () => {
    const { baseElement } = renderWithTheme(
      <AnimatedItem seed="wadus" parentHeight={200} />
    )
    expect(baseElement).toMatchSnapshot()
  })

  it('should render successfully with a path image', () => {
    renderWithTheme(
      <AnimatedItem seed="wadus" parentHeight={200} path="/path/to/image.jpg" />
    )
    expect(screen.getByRole('presentation').firstChild).toHaveStyleRule(
      'background-image: url(/path/to/image.jpg)'
    )
  })

  it('should render successfully with a rotation and a custom size', () => {
    renderWithTheme(
      <AnimatedItem
        seed="wadus"
        parentHeight={200}
        path="/path/to/image.jpg"
        rotate={true}
        size={100}
      />
    )

    expect(screen.getByRole('none')).toHaveStyleRule(
      'animation',
      expect.stringMatching(/(\d+)s rotate-movement linear infinite/)
    )
    expect(screen.getByRole('presentation')).toHaveStyleRule(
      '--itemSize',
      '100px'
    )
  })

  it('should render successfully with a rotation and no path', () => {
    const { baseElement } = renderWithTheme(
      <AnimatedItem seed="wadus" parentHeight={200} rotate={true} />
    )
    expect(baseElement).toMatchSnapshot()
  })

  it('should render successfully with a custom animation', () => {
    const customAnimation = () => ({
      horizontal: css`
        animation: customHorizontal 6s linear infinite;
      `,
      vertical: css`
        animation: customVertical 6s linear infinite;
      `,
      rotate: css`
        animation: customRotate 6s linear infinite;
      `,
    })

    renderWithTheme(
      <AnimatedItem
        seed="wadus"
        parentHeight={200}
        rotate={true}
        path="/test.jpg"
        customAnimation={customAnimation()}
      />
    )

    expect(screen.getByRole('presentation')).toHaveStyleRule(
      'animation',
      'customHorizontal 6s linear infinite'
    )

    const rotationItem = screen.getByRole('none')
    expect(rotationItem.parentElement).toHaveStyleRule(
      'animation',
      'customVertical 6s linear infinite'
    )
    expect(rotationItem).toHaveStyleRule(
      'animation',
      'customRotate 6s linear infinite'
    )
  })
})
