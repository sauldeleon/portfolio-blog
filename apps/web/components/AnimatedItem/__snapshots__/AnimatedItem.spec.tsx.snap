// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnimatedItem should render successfully 1`] = `
.c2 {
  position: absolute;
  width: 100%;
  height: 100%;
  display: grid;
  place-items: center;
  pointer-events: none;
  animation-timing-function: ease-in-out;
  animation-iteration-count: infinite;
  animation-direction: alternate;
  animation-name: TpVAo;
}

.c4 {
  height: var(--itemSize);
  aspect-ratio: 1/1;
  position: absolute;
  pointer-events: none;
}

.c5 {
  pointer-events: all;
  cursor: pointer;
}

.c7 {
  color: #FFDD83;
  animation-name: lhbgW;
  animation-timing-function: linear;
  animation-iteration-count: infinite;
}

.c7 svg {
  width: 100%;
  height: 100%;
}

.c0 {
  --itemSize: 40px;
  list-style: none;
  position: absolute;
  transform: translateX(-60%);
  width: 100%;
  height: 100%;
  pointer-events: none;
  animation-name: gBgZDr;
  animation-timing-function: linear;
  animation-iteration-count: infinite;
}

.c0:hover {
  z-index: 200;
  animation-play-state: paused;
}

.c0:hover .c1,
.c0:hover .c3,
.c0:hover .c6 {
  animation-play-state: paused;
}

@media (width >= 768px) {
  .c0 {
    --itemSize: 55px;
  }
}

<body>
  <div>
    <li
      class="c0"
      data-testid="horizontal-movement"
      style="animation-duration: 14.013358467316657s; animation-delay: 15.815396422171766s;"
    >
      <div
        class="c1 c2"
        data-testid="vertical-movement"
        style="--translateY-begin: -3%; --translateY-end: 31%; animation-duration: 7.388991085926664s;"
      >
        <div
          class="c3 c4"
          data-testid="rotation-movement"
        >
          <a
            aria-label="Google Chrome"
            class="c5"
            href="https://google.com"
            rel="noreferrer"
            target="_blank"
          >
            <div
              class="c6 c7"
              data-testid="color-swapping"
              style="animation-duration: 14.013358467316657s; animation-delay: 15.815396422171766s;"
            >
              <svg />
            </div>
          </a>
        </div>
      </div>
    </li>
  </div>
</body>
`;
